### PR TITLE
Configure Django settings

### LINKED ISSUE #8: Configure Django settings
The Django settings need to be configured for this web app. We should:
- Add the `main` app to `INSTALLED_APPS`
- Refactor the settings into two options, one for development and one for production
- Create the DB directory and adjusts settings accordingly
- Configure static file handling with Whitenoise

Follow [the first 4 commits in this PR](https://github.com/ImperialCollegeLondon/drunc_ui/pull/24/files/f678a41d045042ddf14113a4fd12edb8faf2e2e5) as a guide

### LINKED ISSUE #20: Install and configure Django-stubs
We should include the [`django-stubs`](https://github.com/typeddjango/django-stubs) library to enable type-hinting across the project. This requires some additions to the pyproject.toml (check the setup instructions in the package readme)

### PR DESCRIPTION
**Pull request by barneydobson**
# Description

First pass to mimic the example in the issue

Fixes #8 
Fixes #20 

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (whatever its nature)

## Key checklist

- [ ] All tests pass (eg. `python -m pytest`)
- [ ] The documentation builds and looks OK (eg. `python -m sphinx -b html docs docs/build`)
- [ ] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ authors = [
     { name = "Adrian D'Alessandro", email = "a.dalessandro@imperial.ac.uk" },
 ]
 requires-python = ">=3.12"
-dependencies = ["django"]
+dependencies = ["django", "whitenoise", "django-stubs[compatible-mypy]"]
```
Comment (AdrianDAlessandro) [inline pyproject.toml:9]: The stubs package should only be in the development dependencies. After making that change, the three requirements files should also be updated accordingly with `pip-compile`

**Review by AdrianDAlessandro — CHANGES_REQUESTED** [review]
Looks good, just one change required to the dependencies specification

File: pyproject.toml (hunk 1 above)
Comment (barneydobson) [reply pyproject.toml]: `mypy` seems to fail without it?

Comment (AdrianDAlessandro) [reply pyproject.toml]: Yes, we'll still need it in the `dev` dependencies under `[project.optional-dependencies]` in the `pyproject.toml` file

Comment (barneydobson) [reply pyproject.toml]: I can't seem to edit this from my desktop at home without reforking it - will do on Monday when back in office if it's OK!

Comment (AdrianDAlessandro) [reply pyproject.toml]: Ok, I'll merge this and make that change myself so it gets merged before the end of October

**Review by AdrianDAlessandro — APPROVED** [review]

**Pull request merged**